PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/CreateArmStorageAccountForm.java

```diff
@@ -472,14 +474,16 @@ public void run() {
     private void fillRegions() {
         Subscription subs = (Subscription) subscriptionComboBox.
                 getData(subscriptionComboBox.getText());
-        List<Region> locations = Azure.az(AzureAccount.class).listRegions(subs.getId());
-        for (Region location : locations) {
-            regionComboBox.add(location.getLabel());
-            regionComboBox.setData(location.getLabel(), location);
-        }
-        if (locations.size() > 0) {
-            regionComboBox.select(0);
-        }
+        final List<Region> locations = Azure.az(AzureAccount.class).listRegions(subs.getId());
+        AzureTaskManager.getInstance().runLater(() -> {            
```

Comment:
duplicated run later call for fill regions (Refer Line 457

Besides, shall we make fillRegions to accept `List<Region>`, so that we could get the regions in background and only fill the content in EDT

Reply:
too many such cases, will enlarge the scope of this PR